### ISSUE TITLE
[build] Turn the Standard Library into a regular library

### ISSUE THREAD

**Issue by ejgallego** [coq/stdlib#19]
As discussed many times over the years, it would be great if we could make the standard library a regular library from the point of view of building. We are fairly close, as of today, we only need two things:

- Require users to add `From Coq` when doing a `Require Import foo` where `foo` is a stdlib module.
- Require users to add `From Coq Require Import Prelude` so the dependency on the prelude is now explicit.

This has many advantages, and moreover the changes to libraries are fully backwards compatible.

Unfortunately, it is not easy to give warnings for these particular cases.

When I released jsCoq in 2015, this was the default mode, and was IMHO pretty nice, and it is a mistake that I backtracked on it.

See also the original bug https://github.com/coq/coq/issues/5257 for the deprecation.

**Comment by SkySkimmer** [comment]
Is there anything left to do for this?

**Comment by Zimmi48** [comment]
Indeed, time to close.

**Comment by ejgallego** [comment]
Unless I'm mistaken this is not fully completed, in the sense that there are several very custom hacks in the Rocq codebase to specially support the stdlib.

**Comment by ejgallego** [comment]
> You mean besides the ones that are there for compatibility purposes?

Yup, in particular the ones that turn `Require Import Bar.` into `From Stdlib Require Import Bar.` , IMHO until these are gone, we cannot claim that stdlib is treated as a "regular" library.

**Comment by Zimmi48** [comment]
OK, thanks for clarifying! Indeed, it could be tracked in the Rocq issue tracker (or not tracked at all, as we are probably going to keep these compatibility hacks for a long time).

**Comment by ejgallego** [comment]
> AFAIK the custom hacks are about corelib not stdlib

Are you sure? I still see several stuff, for example `synterp.ml:deprecated_Coq` (also other places do show up)

**Comment by ejgallego** [comment]
> OK, thanks for clarifying! Indeed, it could be tracked in the Rocq issue tracker (or not tracked at all, as we are probably going to keep these compatibility hacks for a long time).

After reading the issue again, indeed it seems to me that both concerns are not solved:
- we still don't require `From Stdlib` to use the Stdlib.
- we still require the prelude.

That's on Coq side tho, so maybe let's reopen and move back to rocq repos.

**Comment by Zimmi48** [comment]
If you think that the present issue should be transferred, rather than a new one opened, feel free to do it.

**Comment by ejgallego** [comment]
I think so, however maybe we should rename it, and maybe split it in two. I'm having trouble coming with a good name tho, "Remove hack to support implicit qualification of Stdlib" ?

Prelude is is `Corelib` now, so IMHO that has become a separate issue.
